lib/portlayer/storage/nfs/target.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nfs
+
+import (
+	"io"
+	"net/url"
+
+	"os"
+)
+
+type MountHandler interface {
+	Mount(target *url.URL) (NFSTarget, error)
+	Unmount(target NFSTarget) error
+}
+
+type NFSTarget interface {
+	//opens a target path in a READONLY context
+	Open(path string) (io.ReadCloser, error)
+
+	//Opens targeted file with the supplied attr.
+	OpenFile(path string, perm os.FileMode) (io.ReadWriteCloser, error)
+
+	//Creates a file, errors out if file already exists. assumes write permissions.
+	Create(path string, perm os.FileMode) (io.ReadWriteCloser, error)
```

Comment:
So for this one it was a nuance of the UnitTests. :-/ Looks like os.OpenFile will return an error if the actual file does not exist. So I moved to having a `Create` here after writing my tests. Let me know if we still want to avoid this. I can change the Target Mock to call os.Create for OpenFile if the file does not exist.